Sources/ApolloCodegenLib/LinkedList.swift

```diff
@@ -71,24 +75,25 @@ public struct LinkedList<T>: ExpressibleByArrayLiteral {
     self.headNode = head
   }
 
+  @_disfavoredOverload
   public init(_ headValue: T) {    
     self.init(head: HeadNode(value: headValue))
   }
 
-  public init(array: [T]) {
-    precondition(!array.isEmpty, "Cannot initialize LinkedList with an empty array. LinkedList must have at least one element.")
-    var segments = array
-    let headNode = HeadNode(value: segments.removeFirst())
+  public init<C: Collection<T>>(_ collection: C) {
```

Comment:
Made this function take in a generic collection instead of only an array. This allows us to pass in another linked list or an ordered set.